### ISSUE TITLE
Would you please strip for release assets?

### ISSUE THREAD

**Issue by abcfy2** [shadowsocks/shadowsocks-rust#276]
I think you can strip for release assets so that thin the tar package.

**Comment by zonyitoo** [comment]
What do you mean? Release packages only contains executable binaries.

**Comment by abcfy2** [comment]
I mean that maybe we could use `strip` to reduce binaries size.

Example:

```
# Before strip
$ file sslocal
sslocal: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, BuildID[sha1]=54d3e530a75fc39515e4c4f56f439c61cbd6133b, with debug_info, not stripped  

$ du -h sslocal
11M     sslocal

# After strip
$ strip sslocal
$ file sslocal
sslocal: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, BuildID[sha1]=54d3e530a75fc39515e4c4f56f439c61cbd6133b, stripped

$ du -h sslocal
8.4M    sslocal
```

You can see after `strip`, `sslocal` file size reduced from 11M to 8.4M.

**Comment by zonyitoo** [comment]
I am going to wait for cargo's strip option: https://github.com/rust-lang/cargo/issues/3483 .

Calling `strip` manually in build script may be nasty in cross builds.

**Comment by zonyitoo** [comment]
```toml
cargo-feature = ["strip"]

[profile.release]
strip = "symbols"
```

This will require a nightly build of `cargo`. I will keep this issue open until this feature is stablized.

**Comment by Kein** [comment]
strip with UPX on top? ;P

**Comment by xnoreq** [comment]
@Kein  Why would you ever want to use UPX? To increase memory usage further because now it also needs to be uncompressed in memory?

**Comment by xnoreq** [comment]
Disabling trust-dns (which is not a performance problem for the client sslocal) and switching from log4rs to stderrorlog (who needs a full-blown logging framework for a socks client?) I get a Windows sslocal binary that's 2.7 MB vs the almost 9MB in the release.

I think this could be reduced further still, but would require going into the dependencies and patching them to make some of their dependencies optional.

**Comment by Kein** [comment]
> I get a Windows sslocal binary that's 2.7 MB vs the almost 9MB in the release.

That is pretty sweet. Got a fork or PR?

**Comment by zonyitoo** [comment]
You could just build it without the `trust-dns` feature.

**Comment by zonyitoo** [comment]
Some guides about how to minimize target files of Rust:

https://github.com/johnthagen/min-sized-rust

**Comment by abcfy2** [comment]
Hi @zonyitoo . Seems latest release still not strip ?

```sh
$ file ss*
sslocal:   ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, with debug_info, not stripped
ssmanager: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, with debug_info, not stripped
ssserver:  ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, with debug_info, not stripped
ssservice: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, with debug_info, not stripped
ssurl:     ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, with debug_info, not stripped
```

**Comment by zonyitoo** [comment]
Yes. The `strip` option will be released in 1.59. The current stable Rust version is 1.58.

The v1.13 release is to make shadowsocks-rust to be built with stable Rust, so I temporary disable the option until 1.59 releases.
